src/nbl/video/COpenGLCommandBuffer.h

```diff
@@ -550,27 +554,75 @@ class COpenGLCommandBuffer final : public IGPUCommandBuffer
         pushCommand(std::move(cmd));
         return true;
     }
-    bool drawIndirect(buffer_t* buffer, size_t offset, uint32_t drawCount, uint32_t stride) override
+    bool drawIndirect(const buffer_t* buffer, size_t offset, uint32_t drawCount, uint32_t stride) override
     {
         SCmd<impl::ECT_DRAW_INDIRECT> cmd;
-        cmd.buffer = core::smart_refctd_ptr<buffer_t>(buffer);
+        cmd.buffer = core::smart_refctd_ptr<const buffer_t>(buffer);
         cmd.offset = offset;
         cmd.drawCount = drawCount;
         cmd.stride = stride;
         pushCommand(std::move(cmd));
         return true;
     }
-    bool drawIndexedIndirect(buffer_t* buffer, size_t offset, uint32_t drawCount, uint32_t stride) override
+    bool drawIndexedIndirect(const buffer_t* buffer, size_t offset, uint32_t drawCount, uint32_t stride) override
     {
         SCmd<impl::ECT_DRAW_INDEXED_INDIRECT> cmd;
-        cmd.buffer = core::smart_refctd_ptr<buffer_t>(buffer);
+        cmd.buffer = core::smart_refctd_ptr<const buffer_t>(buffer);
         cmd.offset = offset;
         cmd.drawCount = drawCount;
         cmd.stride = stride;
         pushCommand(std::move(cmd));
         return true;
     }
 
+    inline bool drawMeshBuffer(const nbl::video::IGPUMeshBuffer* meshBuffer) override
```

Comment:
split into a `drawMeshBuffer_pre` and `post`, move up to IGPUCommandBuffer (so that Vulkan may benefit from the impl as well)